options.go

```diff
@@ -120,8 +121,29 @@ func HashRingConfig(c *hashring.Configuration) Option {
 // automatically.
 func Logger(l log.Logger) Option {
 	return func(r *Ringpop) error {
-		r.logger = l
-		r.log = l
+		if r.moduleLogger == nil {
+			r.moduleLogger = rutil.NewModuleLogger(l)
+		} else {
+			r.moduleLogger.SetLogger(rutil.RootModule, l)
+		}
+		return nil
+	}
+}
+
+// ModuleLevel is used to configure different log levels per module. Messages
+// with a level less than minLevel are silenced. By default, an unconfigured
+// module outputs all messages.
+func ModuleLevel(name string, minLevel rutil.Level) Option {
```

Comment:
I feel like maybe the logging configuration should be passed as a struct.

Functional options give us a couple of advantages:

* We can take in runtime objects as options (e.g. logger created at runtime)
* Options can be optional

In the case of some configuration, I think we should just take a struct if we're only receiving config values. For example, note the `HashRingConfig` option which takes a `hashring.Configuration` type.

It will help us to have a more typed configuration when we want to take configuration from other places later on (e.g. from admin endpoints).

The way I imagine the configuration for the logging is something like:

Logging config for each logger:
```
package modulelogger

type LoggerConfig struct {
	Level int
}
```

Ringpop would have its own logging configuration, with an individual log config for each logger in the project:
```
package ringpop

type LoggerConfiguration struct {
	swim    modulelogger.LoggerConfig
	replica modulelogger.LoggerConfig
	gossip  modulelogger.LoggerConfig
}
```

The logging configuration can be passed in during Ringpop's construction, via a `LogConfig` option:
```
func LogConfig(c *LoggerConfiguration) Option {
	return func(r *Ringpop) error {
		...
	}
}
```